File: L007/inverseArr.cpp
#include<iostream>
#include<vector>
using namespace std;

void inverse(vector<int>& a){

        vector<int>inv(a.size(),0);
        for(int i=0;i<a.size();i++){
            int val = a[i];
            inv[val] = i;
        }


        for(int ele:inv) cout<<ele<<endl;
    }
    int main(){

        int n;
        cin>>n;

        vector<int>a(n,0);
        for(int i=0;i<a.size();i++){
            cin>>a[i];
        }

        inverse(a);
        
        return 0;
    }

File: L007/inverseArr.java
import java.util.Scanner;

public class inverseArr{

    public static Scanner sc = new Scanner(System.in);
    public static void inverse(int[] a){

        int[] inv = new int[a.length];
        for(int i=0;i<a.length;i++){
            int val = a[i];
            inv[val] = i;
        }


        for(int ele:inv) System.out.println(ele);
    }
    public static void main(String[] args){

        int n = sc.nextInt();

        int[] a = new int[n];
        for(int i=0;i<a.length;i++){
            a[i] = sc.nextInt();
        }

        inverse(a);
    }
}